cv-server/src/MockerWebcamREST/mocker_webcam.py

```diff
@@ -0,0 +1,241 @@
+            self.replay_btn["state"] = "normal"
+            self.open_file_btn["state"] = "normal"
+            self.play_btn["state"] = "normal"
+            self.stop_btn["state"] = "normal"
+            self.video_capture.reset()
+            self.webcam = False
+        else:
+            self.toggle_webcam_btn.config(background = "red")
+            self.replay_btn["state"] = "disabled"
+            self.open_file_btn["state"] = "disabled"
+            self.play_btn["state"] = "disabled"
+            self.stop_btn["state"] = "disabled"
+            self.video_capture.set(0)
+            self.resize_video_canvas()
+            self.playing = True
+            self.webcam = True
+
+    def play_video(self):
+        if self.video_capture.video != None:
+            self.playing = True
+    
+    def stop_video(self):
+        if self.video_capture.video != None:
+            self.playing = False
+
+    def replay_video(self):
+        if self.video_capture.video != None:
+            self.playing = True
+            self.video_capture.set(self.video_capture.current)
+
+    def select_file(self):
+        filename = filedialog.askopenfile(
+            title='Open a video file',
+            initialdir = os.getcwd(),
+            filetypes = [('MP4 files', '*.mp4')]
+        )
+        if filename != None:
+            self.video_capture.set(filename.name)
+            self.resize_video_canvas()
+            self.playing = False
+            filename.close()
+
+    def resize_video_canvas(self):
+        self.video_canvas.config(
+            width = self.video_capture.width,
+            height = self.video_capture.height,
+            background = "black"
+        )
+
+class VideoCapture():
+    def __init__(self) -> None:
+        self.video = None
+        self.current = ""
+        self.width = 0
+        self.height = 0
+    
+    def set(self, path: str) -> None:
+        if self.video != None:
+            self.video.release()
+        self.video = cv2.VideoCapture(path)
+        if not self.video.isOpened():
+            raise ValueError("Unable to open video source", path)
+        self.width = self.video.get(cv2.CAP_PROP_FRAME_WIDTH)
+        self.height = self.video.get(cv2.CAP_PROP_FRAME_HEIGHT)
+        self.current = path
+
+    def reset(self):
+        self.video = None
+
+    def get_frame(self):
+        if self.video != None and self.video.isOpened():
+            ret, frame = self.video.read()
+            if ret == True:
+                return (ret, cv2.cvtColor(frame, cv2.COLOR_BGR2RGB))
+            else:
+                return (ret, None)
+        else:
+            return (False, None)
+        
+    def __del__(self):
+        if self.video != None and self.video.isOpened():
+            self.video.release()
+
+class Frame(BaseModel):
+    image_base64: str
+
+class MockerCaptureCameraServer:
+    def __init__(self, mocker) -> None:
+        self.UNITY_SERVER_URL = "http://localhost:7000"
```

Comment:
W sumie to takie rzeczy powinniśmy parametryzować, ale to se zobaczymy w przyszłości. Może dodamy jsona w LocalLow i się będziemy do niego odwoływać, by takich danych jak porty nie hardkodować